### PR TITLE
feat: add configurable bootstrap to client

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

File: src/client/mod.rs [hunk 1]
```diff
@@ -309,8 +317,14 @@ pub async fn attempt_bootstrap(
     qp2p_config: &QuicP2pConfig,
     keypair: Arc<Keypair>,
     notification_sender: UnboundedSender<ClientError>,
+    bootstrap_nodes: Option<HashSet<SocketAddr>>,
 ) -> Result<ConnectionManager, ClientError> {
     let mut attempts: u32 = 0;
+    let mut qp2p_config = qp2p_config.clone();
+
+    if let Some(contacts) = bootstrap_nodes {
+        qp2p_config.hard_coded_contacts = contacts;
```
Comment (lionel1704) [inline src/client/mod.rs:326]: Is this meant to replace all the contacts in the qp2p_config ? Perhaps this should be extend ?

Comment (lionel1704) [reply src/client/mod.rs:326]: If we change this to extend() then we can remove the Option as well. So just pass Default::default() instead of None.

Comment (joshuef) [reply src/client/mod.rs:326]: Hmm, I'm not sure. I'd guess if you're passing in something you may want to override? (what if trying to override a broken network as default eg)

Comment (joshuef) [reply src/client/mod.rs:326]: >we can remove the Option as well. So just pass Default::default() instead of None.
Where do you mean sorry?

Comment (bochaco) [reply src/client/mod.rs:326]: I think we should keep it simple and just allow for overriding, and therefore we need the `Option<...>`. Although, ar we ok with having a `HashSet` as in the API's args? I always tend to use the most primitive data types for API args...

Comment (joshuef) [reply src/client/mod.rs:326]: I've just rolled with what was existing for BootstrapConfig here. I did read up on the diff w/ vec, and it was preventing duplicates. which could be useful here

**Review by bochaco — APPROVED** [review]

**Pull request merged**